internal/device/rapl_sysfs_power_meter.go

```diff
@@ -146,13 +146,52 @@ func (r *raplPowerMeter) Zones() ([]EnergyZone, error) {
 		stdZoneMap[key] = zone
 	}
 
-	r.cachedZones = make([]EnergyZone, 0, len(stdZoneMap))
-	for _, zone := range stdZoneMap {
-		r.cachedZones = append(r.cachedZones, zone)
-	}
+	// Group zones by name for aggregation
+	r.cachedZones = r.groupZonesByName(stdZoneMap)
 	return r.cachedZones, nil
 }
 
+// groupZonesByName groups zones by their base name and creates AggregatedZone
+// instances when multiple zones share the same name (multi-socket systems)
+func (r *raplPowerMeter) groupZonesByName(stdZoneMap map[zoneKey]EnergyZone) []EnergyZone {
+	// Group zones by base name (e.g., "package", "dram")
+	zoneGroups := make(map[string][]EnergyZone)
+
+	for key, zone := range stdZoneMap {
+		zoneGroups[key.name] = append(zoneGroups[key.name], zone)
+	}
+
+	// Create aggregated zones for duplicates, keep single zones as-is
+	var result []EnergyZone
+	for name, zones := range zoneGroups {
+		if len(zones) == 1 {
+			// Single zone - use as-is
+			result = append(result, zones[0])
+			continue
+
+		}
+
+		// Multiple zones with same name - create AggregatedZone
```

Comment:
is there an option to not aggregate in some rare case?

Reply:
Good question and I think we will need that when/if kepler supports measuring individual socket consumption and is able to attribute that to process running on those sockets. 

Or are you talking about some other requirement that we may have at the moment?

Reply:
some reason we cant think of now. the default can be to aggregate, but if needed a config change (unexposed perhaps) will stop the aggregation.